### ISSUE TITLE
Can't Drag n' Drop more than two images into editor From v. 4.4.0 - 4.7.2

### ISSUE THREAD

**Issue by donShakespeare** [tinymce/tinymce#4055]
**Expected behaviour**
Drag plenty images from PC into editor, and see the exact number in the editor... (9, 10, even 20 images)

**Current: behaviour**
Only two will show in the editor ... all versions since TinyMCE  4.4.0 (in latest Chrome)

**Replicate in Fiddle**
```
tinymce.init({
    selector: "textarea",
    plugins: "image imagetools paste",
    paste_data_images: true,
});
```

**Observation**
When a valid upload script is provided, all 10 of 10 dropped images get uploaded... so that means all 10 got dragged in ...
But only 2 will ever show up in the editor.

**Hunch**
Something about `selection.setRng` or images killing off previously auto-selected images

See https://community.tinymce.com/communityQuestion?id=90661000000QdDEAA0

**Comment by donShakespeare** [comment]
As per the title of this issue, I guess you can drag and drop, you just won't see the remaining of your images.

I just noticed that ...
In a list of 10 dropped images ..

1. the first image is always finally inserted into the editor
2. then the second image is the one the mouse pointer dragged the whole selection from

If you dragged 10 images, and the cursor was on the 9th, then the first and ninth image will finally be inserted into the editor.

**Comment by LowSociety** [comment]
This issue seems related to the fact that the range is set for each image [here](https://github.com/tinymce/tinymce/blob/7b10860a76a07db43ad0677bee4b2db64dbddef0/src/plugins/paste/main/ts/core/Clipboard.ts#L143). I've never looked at the source for tinymce before but since the images are passed in a loop [here](https://github.com/tinymce/tinymce/blob/7b10860a76a07db43ad0677bee4b2db64dbddef0/src/plugins/paste/main/ts/core/Clipboard.ts#L201), does the range have to be set on every iteration?

**Comment by barteksekula** [comment]
@LowSociety you were right, thanks a lot for your comment ; ) Your suggestion did help. https://github.com/tinymce/tinymce/pull/4406

**Comment by pvldruzhinin** [comment]
Same issue on 4.8.2 version: here is a video http://take.ms/8H7WL

**Comment by donShakespeare** [comment]
I don't think it is fixed yet. Still having the issue.

**Comment by lnewson** [comment]
Sorry for the really late update here, but this was fixed in TinyMCE 4.9.1 via https://github.com/tinymce/tinymce/commit/bba2b16dd539acd0a9f5bfa27830c5e6d11cabc9. I've tested that version and the latest 5.x version and it works in my testing. As such I'm going to close this, but if you're still experiencing issues with TinyMCE 4.9.1 or higher please let us know.
